### DISCUSSION TITLE
how to create multiple independent ganache instances in nodejs?

### DISCUSSION THREAD

**Discussion by mookim-eth** [ConsenSys-archive/ganache#4552] [Q&A] [answered]
i tried using `let provider = ganache.provider(options);` with different `options.fork.blockNumber`, but they use the same underlying blockchain instead of creating a new instance.

How to create different independent chains in a nodejs process?

**Comment by davidmurdoch** [comment]
I don't think they do. Can you provide example code?

**Reply by mookim-eth** [reply, marked as answer]
thanks for your reply, I found the issue is caused by my `database.dbPath` options, setting the same folder will re-use the old blockchain state instead of creating a new one
